rs/execution_environment/src/execution_environment/tests/canister_snapshots.rs

```diff
@@ -1439,54 +1439,6 @@ fn load_canister_snapshot_fails_snapshot_not_found() {
     assert!(test.state().canister_state(&canister_id).is_some());
 }
 
-#[test]
-fn load_canister_snapshot_fails_snapshot_does_not_belong_to_canister() {
```

Comment:
Test is moved in another file and repurposed to check the opposite condition. The move is not strictly speaking necessary but it makes it easier to re-use the function that allows to check for two snapshots being equal as it requires using a state machine test (instead of execution test).

Reply:
Update: test was changed to check the condition that was additionally added for the sender being controller of the canister which the snapshot belongs to (apparently, state machine tests do not allow you to change the user id that sends requests easily which makes checking this easy only with execution tests).

Reply:
> state machine tests do not allow you to change the user id that sends requests easily

one can specify the sender via `StateMachine::execute_ingress_as`.